src/component/parser/json/converter/DiagramConverter.ts

```diff
@@ -45,23 +59,35 @@ export default class DiagramConverter extends AbstractConverter<BpmnModel> {
         continue;
       }
 
+      const pool = this.deserializeShape(shape, (bpmnElement: string) => findProcessElement(bpmnElement));
+      if (pool) {
+        convertedShapes.pools.push(pool);
+        continue;
+      }
+
       // TODO error management
-      console.log('Not possible to find model element with id ' + shape.bpmnElement);
+      console.log('Shape json deserialization: unable to find bpmn element with id %s', shape.bpmnElement);
     }
 
     return convertedShapes;
   }
 
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   private deserializeShape(shape: any, findShapeElement: (bpmnElement: string) => ShapeBpmnElement): Shape | undefined {
-    const bpmnElement = findShapeElement(shape.bpmnElement);
+    let bpmnElement = findShapeElement(shape.bpmnElement);
     if (bpmnElement) {
-      const id = shape.id;
-
       const jsonConvert: JsonConvert = JsonParser.getInstance().jsonConvert;
       const bounds = jsonConvert.deserializeObject(shape.Bounds, Bounds);
 
-      return new Shape(id, bpmnElement, bounds);
+      if (bpmnElement.parentId) {
+        const participant = findProcessRefParticipantByProcessRef(bpmnElement.parentId);
+        if (participant) {
+          // clone to avoid modifying the reference
+          bpmnElement = new ShapeBpmnElement(bpmnElement.id, bpmnElement.name, bpmnElement.kind, participant.id);
```

Comment:
I don't understand why we don't want to modify the reference.

If the first time, we have the processRef as parentId, we could it.
And the next time no need to modify it.
It avoid to have a lot of object.

Reply:
I guess when I implemented that I was afraid of side effect, but as we have no test to show that, I now think cloning is useless (at least for now)
So I will remove the clone, and directly modify the ref as you suggest @csouchet

Reply:
@csouchet done via c45fb73